### PR TITLE
Fix: MAASENG-1332 - stack form buttons on smaller screens

### LINKED ISSUE #3334: Form buttons get squashed on smaller screen sizes
This is only an issue with the forms that include a "Save and add another" button

Screen Shot 2021-11-25 at 1 54 05 pm
Screen Shot 2021-11-25 at 1 54 12 pm

_Originally posted by @huwshimi in https://github.com/canonical-web-and-design/maas-ui/pull/3331#pullrequestreview-815517294_

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done

- Stacked form buttons on smaller screens (< 1036px).

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to `/MAAS/r/devices`
- Click the `Add Device` button
- Examine the form buttons at the base on different screen sizes

## Fixes

Fixes: # .
closes #3334 
[https://warthogs.atlassian.net/browse/MAASENG-1332](https://warthogs.atlassian.net/browse/MAASENG-1332)

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

image

### CODE REVIEW

**Review by ndv99 — COMMENTED** [review]
This looks good to me - code is solid and the buttons respond nicely. I'm not certain about this from a design perspective though - @amylily1011 @petermakowski what do you think?

File: src/app/base/components/FormikFormButtons/_index.scss [hunk 1]
```diff
@@ -43,5 +43,18 @@
         width: 100%;
       }
     }
+
+    @media only screen and (max-width: $breakpoint-large) {
+      .formik-form-buttons__container {
+        display: flex;
+        flex-direction: column-reverse;
```
Comment (petermakowski) [inline src/app/base/components/FormikFormButtons/_index.scss:50]: I appreciate your initiative here, but having thought about this a bit more I'm not sure this is needed.

It's not clear if the reverse button order on mobile by default is best from UX perspective. But also this has wide implications - you'd need to check it's appropriate and works well for every case where these buttons are used.

**Review by petermakowski — APPROVED** [review]
Looks great, but think we might need to reconsider button's order.

**Comment by amylily1011** [comment]
If this is for mobile, it is less error prone to place the most prominent button in the bottom. Meaning the green one should be the last one and `Cancel` should be the first one. This is because when operating on mobile, the nearest interaction to your thumb should be the primary interaction which is more likely to be close to bottom of the device.

File: src/app/base/components/FormikFormButtons/_index.scss (hunk 1 above)
Comment (Jay-Topher) [reply src/app/base/components/FormikFormButtons/_index.scss:50]: Okay, I'll make the change now

**Comment by Jay-Topher** [comment]
Here's what it looks like now 

image

**Review by petermakowski — APPROVED** [review]

**Pull request merged**